feat(router-devtools-core): add navigation link to route

scarf005 opened the pull request:
https://github.com/user-attachments/assets/8abac796-e54f-40f9-a725-9aa3e52e5810

- resolves #4039
- dynamic routes and search params aren't supported.

SeanCassiere reviewed (changes requested):
I'd like to have some code comments added in for these functions/variables:

- `routeKind` -  Just comment in what each of the "kinds" mean.
- `hasValidParamValueInMatch` - Purpose of the function and comments at each exit statements and what that means.
- `navigateToPath` - Explain why there are different conditions for the `string` being returned.

Overall, I like this PR and want to get this landed in.

SeanCassiere commented:
I'm trying to figure out why CI is failing, but I do not have a lot of time to look into it at the moment.

@scarf005 please request a re-review from me once you've added in the changes I asked for. I'll run the tests locally and see if there's actually an "issue" with these changes that are somehow invisible at first-glance, or if its just the CI pipeline being difficult.

scarf005 commented:
@SeanCassiere hi, i refactored the code again and was able to simplify most of the logic. now `routeKind`, `hasValidParamValueInMatch`, and `navigateToPath` is all merged into `navigationTarget`, and i've added comments for each conditions.

scarf005 commented on packages/router-devtools-core/src/BaseTanStackRouterDevtoolsPanel.tsx:
```diff
@@ -126,6 +149,45 @@ function RouteComp({
+    if (!route.path) return undefined // no path to navigate to
+
+    // flatten all params in the router state, into a single object
+    const allParams = matches()
+      .flatMap((m) => m.params)
+      .reduce((prev, curr) => {
+        const keys = Object.keys(curr)
+        for (const key of keys) {
+          if (prev[key] === undefined) {
+            prev[key] = curr[key]
+          }
+        }
+        return prev
+      }, {})
```
```suggestion
    const allParams = Object.assign({}, ...matches().reverse().flatMap(m => m.params))
```

@SeanCassiere how about this apporach?

SeanCassiere replied:
Had to make a few modifications, but the essential gist of your suggestion has been applied.

SeanCassiere commented:
@scarf005 thank you for picking this up and contributing!
